### ISSUE TITLE
Unused using directive is not reported.

### ISSUE THREAD

**Issue by JosefPihrt** [dotnet/roslyn#18673]
**Version Used**: Visual Studio 2017 15.1.0+26403.3

**Code to Reproduce**:
```csharp
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.CSharp; // this using is not used but it is not reported as unused.

namespace N
{
    public static class C
    {
        public static void M(this SyntaxNode root, SyntaxTrivia oldTrivia, SyntaxTrivia newTrivia)
        {
            SyntaxNode newRoot = root.ReplaceTrivia(oldTrivia, newTrivia);
        }
    }
}
```
**Expected Behavior**:
using directive `Microsoft.CodeAnalysis.CSharp` is not used so it should be reported.

**Comment by Pilchie** [comment]
Starting with compiler, as we just report the results of the API call.

**Comment by gafter** [comment]
The extension method `Microsoft.CodeAnalysis.CSharp.CSharpExtensions.ReplaceTrivia(this SyntaxToken token, ...)` is a member of the method group referenced by the expression `root.ReplaceTrivia`, even though it is not selected by overload resolution as the method being invoked. The compiler thus does need to (and does) traverse through that using directive to bind the `ReplaceTrivia` invocation.

Thus the compiler is pedantically technically correct (which is the best kind of correct, right?).
